From: "Dmitry V. Levin" <ldv@altlinux.org>
Subject: uapi: fix linux/sysctl.h userspace compilation errors

Include <stddef.h> (guarded by #ifndef __KERNEL__) to fix the following
linux/sysctl.h userspace compilation errors:

/usr/include/linux/sysctl.h:38:2: error: unknown type name 'size_t'
  size_t *oldlenp;
/usr/include/linux/sysctl.h:40:2: error: unknown type name 'size_t'
  size_t newlen;

This also fixes userspace compilation of uapi headers that include
linux/sysctl.h, e.g. linux/netfilter.h.

Link: http://lkml.kernel.org/r/20170222230652.GA14373@altlinux.org
Signed-off-by: Dmitry V. Levin <ldv@altlinux.org>
Cc: Alexey Dobriyan <adobriyan@gmail.com>
Cc: "Eric W. Biederman" <ebiederm@xmission.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 include/uapi/linux/sysctl.h |    4 ++++
 1 file changed, 4 insertions(+)

diff -puN include/uapi/linux/sysctl.h~uapi-fix-linux-sysctlh-userspace-compilation-errors include/uapi/linux/sysctl.h
--- a/include/uapi/linux/sysctl.h~uapi-fix-linux-sysctlh-userspace-compilation-errors
+++ a/include/uapi/linux/sysctl.h
@@ -27,6 +27,10 @@
 #include <linux/types.h>
 #include <linux/compiler.h>
 
+#ifndef __KERNEL__
+#include <stddef.h>		/* For size_t. */
+#endif
+
 #define CTL_MAXNAME 10		/* how many path components do we allow in a
 				   call to sysctl?   In other words, what is
 				   the largest acceptable value for the nlen
_
